src/components/views/directory/RoleDirectoryDetail.tsx

```diff
@@ -0,0 +1,270 @@
+            method: "GET"
+        }).then(res => res.json())
+            .then((response) => {
+                let displayName = response.entry.displayName;
+                this.setState({ displayName: displayName })
+            })
+        let fullName = this.state.displayName;
+        return fullName;
+    }
+
+    // The phone, email address info was already converted to
+    // string bundle it needs to be changed back to array then be formatted properly
+    // A utility function that converts an string containing pair value into array
+    // converts a given value number into landline or mobile number.
+    // returns 'Landline' or 'mobile' number with correct text which can be presentable.
+    getFormattedPhoneNumber(value) {
+        if (value == undefined || value == null) {
+            return value;
+        }
+        const phoneNumber = value.map((value, index) => {
+            let newPhoneNumber = new Array(value);
+            // Generates landline or mobile number
+            let phoneNumberType = newPhoneNumber.map((value) => value.type);
+            //find actual phone number digit
+            let phoneNumber = newPhoneNumber.map((value) => value.value);
+            const phoneFormat = phoneNumberType.toString().charAt(0).toUpperCase() + phoneNumberType.toString().slice(1).toLowerCase() + ' - ';
+            return <div key={index}><span>{phoneFormat}</span><span>{phoneNumber}</span></div>
+        });
+
+        return phoneNumber;
+
+    }
+
+    //format role id columns
+    getFormattedRoleIds(key: string) {
+        if (key == undefined || key == null) {
+            return key;
+        }
+        //if values are listed in enum format from there first
+        const formattedValuesWithEnums = getRoleEnumValues(key);
+        // if role ids are not listed in enums format using followings.
+        const formattedValues = () => {
+            key = key.charAt(0).toUpperCase() + key.slice(1); // turn first letter to capital
+            key = key.replace(/([A-Z]+)/g, ",$1").replace(/^,/, "");
+            return key.split(',').join(' ');
+        }
+        const finalFormattedValues = formattedValuesWithEnums || formattedValues();
+        return finalFormattedValues;
+    }
+
+    getFormattedRoleIDTextLabel(id: string, headerElement) {
+        if (id == undefined || id == null) {
+            return id;
+        } else {
+            id = headerElement[headerElement.indexOf(id)];
+            return this.getFormattedRoleIds(id);
+        }
+    }
+
+    getNameFromEmail(email: string): string {
+        // checks email addresses and splits first and last names accordingly
+        if (!email || (email === undefined) || (email.indexOf('@') === -1)) return null;
+        if (email[0] === '@') {  // remove @ initial (valid in matrix id)
+            email = email.substring(1);
+        }
+        let firstName = email.split('.')[0];
+        firstName = firstName.charAt(0).toUpperCase() + firstName.slice(1);
```

Comment:
Please centralise this repeated logic

Reply:
I have moved all of these functions into utils/ folder which is default matrix-react-sdk folder for storing sharable utility functions by their type and usage. Also took little bit to refactor one of the function to make it reusable by removing html content. Now it looks smooth. Didnt have time before but done now. All changes pushed.